cfme/infrastructure/host.py

```diff
@@ -559,3 +566,36 @@ def find_quadicon(host, do_not_navigate=False):
             return quadicon
     else:
         raise HostNotFound("Host '{}' not found in UI!".format(host))
+
+
+def _method_setup(host_names, provider_crud=None):
+    """ Reduces some redundant code shared between methods """
+    if isinstance(host_names, basestring):
+        host_names = [host_names]
+
+    if provider_crud:
+        provider_crud.load_all_provider_nodes()
+    else:
+        navigate_to(Host, 'All')
+    if paginator.page_controls_exist():
+        paginator.results_per_page(1000)
+    for host_name in host_names:
+        sel.check(Quadicon(host_name, 'host').checkbox())
+
+
+def wait_for_host_state_change(host_name, desired_state, timeout=300, provider_crud=None):
```

Comment:
If anything this should be a host_method, we shouldn't be passing around strings for names anymore, but objects.